### PR TITLE
tests: follow up to clean up go-check staff

### LINKED ISSUE #27874: migrate test-infra to testify for `tests` folder
* global_kill_test.go
* graceshutdown_test.go
* readonly_test.go

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

### What problem does this PR solve?

Issue Number: close #27874 

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/globalkilltest/main_test.go [hunk 1]
```diff
@@ -15,13 +15,13 @@
...
 	"testing"
 
 	"github.com/pingcap/tidb/util/testbridge"
-	"go.uber.org/goleak"
 )
 
 func TestMain(m *testing.M) {
 	testbridge.WorkaroundGoCheckFlags()
-	goleak.VerifyTestMain(m)
+	os.Exit(m.Run())
```
Comment (xhebox) [inline tests/globalkilltest/main_test.go:26]: So drop the leak test for kill?

File: tests/readonlytest/readonly_test.go [hunk 2]
```diff
@@ -142,6 +137,7 @@ func TestRestrictionWithConnectionPool(t *testing.T) {
 			} else {
 				done <- false
 			}
+			return
```
Comment (xhebox) [inline tests/readonlytest/readonly_test.go:140]: That is un-common. Can we change to a pattern like:

```
if err != nil {
   if err.Error() ==
   else
   return
}
```

File: tests/globalkilltest/main_test.go (hunk 1 above)
Comment (tisonkun) [reply tests/globalkilltest/main_test.go:26]: We don't check it previously and it causes more time to dig out whether to handle these leaks. I don't have such spare time.

File: tests/readonlytest/readonly_test.go (hunk 2 above)
Comment (tisonkun) [reply tests/readonlytest/readonly_test.go:140]: Let me take a closer look.

Comment (tisonkun) [reply tests/readonlytest/readonly_test.go:140]: updated.

**Review by xhebox — APPROVED** [review]

**Review by zwj-coder — APPROVED** [review]

**Pull request merged**